display_list/display_list_canvas_unittests.cc

```diff
@@ -229,106 +238,209 @@ class BoundsTolerance {
-      int width = kTestWidth,
-      int height = kTestHeight) const {
-    sk_sp<SkSurface> surface =
-        SkSurface::MakeRaster(info_.makeWH(width, height));
-    surface->getCanvas()->clear(bg);
-    return std::make_unique<RenderSurface>(surface);
+  static RenderEnvironment MakeN32(const DlSurfaceProvider* provider) {
+    return RenderEnvironment(provider, PixelFormat::kN32Premul_PixelFormat);
   }
 
   void init_ref(CvRenderer& cv_renderer, DlColor bg = DlColor::kTransparent()) {
-    init_ref([=](SkCanvas*, SkPaint&) {}, cv_renderer,
-             [=](DisplayListBuilder&) {}, bg);
+    init_ref(kEmptyCvSetup, cv_renderer, kEmptyDlRenderer, bg);
   }
 
   void init_ref(CvSetup& cv_setup,
                 CvRenderer& cv_renderer,
                 DlRenderer& dl_setup,
                 DlColor bg = DlColor::kTransparent()) {
-    ref_canvas()->clear(bg);
+    CvRenderJob job = {
+        .bg = bg,
+        .cv_setup = cv_setup,
+        .cv_render = cv_renderer,
+    };
+    ref_result_ = getResult(job);
     dl_setup(ref_attr_);
-    SkPaint paint;
-    cv_setup(ref_canvas(), paint);
-    ref_matrix_ = ref_canvas()->getTotalMatrix();
-    ref_clip_ = ref_canvas()->getDeviceClipBounds();
-    cv_renderer(ref_canvas(), paint);
-    ref_pixmap_ = ref_surface_->pixmap();
   }
 
-  const SkImageInfo& info() const { return info_; }
-  SkCanvas* ref_canvas() { return ref_surface_->canvas(); }
+  template <typename J>
+  std::unique_ptr<RenderResult> getResult(J& job) const {
+    auto surface = getSurface(job.width, job.height);
+    FML_DCHECK(surface != nullptr);
+    auto canvas = surface->getCanvas();
+    canvas->clear(job.bg);
+
+    int restore_count = canvas->save();
+    canvas->scale(job.scale, job.scale);
+    job.render(canvas);
+    canvas->restoreToCount(restore_count);
+
+    canvas->flush();
+    surface->flushAndSubmit(true);
+    return std::make_unique<RenderResult>(surface, job.setup_matrix,
+                                          job.setup_clip_bounds);
+  }
+
+  std::unique_ptr<RenderResult> getResult(sk_sp<DisplayList> dl) const {
+    DlRenderJob job = {
+        .display_list = std::move(dl),
+    };
+    return getResult(job);
+  }
+
+  const DlSurfaceProvider* provider() const { return provider_; }
+  bool valid() const { return provider_->supports(format_); }
+  const std::string backend_name() const { return provider_->backend_name(); }
+
+  PixelFormat format() const { return format_; }
   const DisplayListBuilder& ref_attr() const { return ref_attr_; }
-  const SkMatrix& ref_matrix() const { return ref_matrix_; }
-  const SkIRect& ref_clip_bounds() const { return ref_clip_; }
-  const SkPixmap* ref_pixmap() const { return ref_pixmap_; }
+  const RenderResult* ref_result() const { return ref_result_.get(); }
 
  private:
-  explicit RenderEnvironment(const SkImageInfo& info) : info_(info) {
-    ref_surface_ = MakeSurface();
+  static bool matches(const DlSurfaceInstance* surface, int w, int h) {
```

Comment:
Nit: `SizeMatches` maybe (go/fluttercppl#Function_Names)? Though I suppose mixing this style and Skia style is common here so perhaps fine for consistency.

Reply:
At one point they were constructed as needed and could be null so the test was more than just the size, but then I went with just hand-coding 2 different sizes as I know that's all the tests use and so this function isn't really useful any more.

Reply:
Changed names to `surface_1x_` and `surface_2x_` to make them more obvious.